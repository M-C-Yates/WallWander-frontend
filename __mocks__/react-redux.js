jest.mock('react-redux', () => {
	return {
		useDispatch() {
			return { dispatch: jest.fn() };
		},
		useSelector() {
			return {
				id: '12345',
				email: 'foo@bar.com',
				name: 'foobar',
				token: '123abc',
				nsfw: false
			};
		}
	};
});
