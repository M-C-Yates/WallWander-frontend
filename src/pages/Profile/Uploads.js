import React, { useEffect, useState } from "react";
import axios from "axios";

import { useDispatch, useSelector } from "react-redux";

import Navbar from "../../components/Navbar";
import ProfileNav from "./ProfileNav";
import { WallpaperList } from "../../components/wallpaper/WallpaperList";
import { getImages } from "../../actions/image";
import { baseUrl } from "../../App";

export const Uploads = props => {
  const [images, setImages] = useState([]);
  const [paginationData, setPaginationData] = useState({});

  const user = useSelector(state => state.auth);

  const dispatch = useDispatch();

  useEffect(() => {
    const fetchData = async () => {
      const { data } = await axios.get(
        `${baseUrl}/images/uploads/${props.id}?limit=15&page=0`
      );
      const images = data.data;
      setImages(images);
      setPaginationData({
        lastPage: data.last_page,
        currentPage: data.current_page
      });
      dispatch(getImages(images));
      return;
    };
    fetchData();
    return;
  }, [dispatch, props.id, user.id]);

  return (
    <div className="flex flex-col items-center">
      <Navbar />
      <ProfileNav uri={props.uri} id={props.id} />
      <WallpaperList
        images={images}
        setImages={setImages}
        pageChangeUrl={`${baseUrl}/images/uploads/${props.id}`}
        paginationData={paginationData}
      />
    </div>
  );
};

export default Uploads;
