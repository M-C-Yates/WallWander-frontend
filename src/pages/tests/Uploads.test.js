import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { useDispatch } from 'react-redux';

import Uploads from '../Profile/Uploads';

afterEach(cleanup);

describe('Uploads', () => {
	it('should match snapshot', () => {
		const { asFragment } = render(<Uploads />);
		expect(asFragment).toMatchSnapshot();
	});
});
