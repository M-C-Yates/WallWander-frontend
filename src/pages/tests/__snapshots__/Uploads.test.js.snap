// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Uploads should match snapshot 1`] = `[Function]`;
